### PR TITLE
List: Add support for defining which fields are included in the output

### PR DESCRIPTION
**Pull request by philipjohn**
Allows users to define a list of DB fields that should be included in the output, defaulting to ID, post_title and post_parent.

P.s. I haven't actually tested this yet :smile:

### CODE REVIEW

File: revisions-cli.php [hunk 1]
```diff
@@ -76,10 +80,62 @@ public function dump( $args = array(), $assoc_args = array() ) {
...
+				'comment_count',
+			];
+
+			// Don't allow fields that aren't in the above whitelist.
+			$excluded_fields = array_diff( $assoc_args['fields'], $allowed_fields );
+			if ( ! empty( $excluded_fields ) ) {
+				WP_CLI::error( 'Invalid values provided in the fields argument.' );
+			}
+
+			$fields = implode( ',', $assoc_args['fields'] );
```
Comment (trepmal) [inline revisions-cli.php]: `$fields` is only imploded if `--fields` is passed

File: revisions-cli.php [hunk 2]
```diff
@@ -76,10 +80,62 @@ public function dump( $args = array(), $assoc_args = array() ) {
...
+		}
+
 		global $wpdb;
 		if ( ! empty( $assoc_args['post_id'] ) ) {
 
-			$revs = $wpdb->get_results( $wpdb->prepare( "SELECT ID, post_title, post_parent FROM $wpdb->posts WHERE post_parent = %d", $assoc_args['post_id'] ) );
+			$revs = $wpdb->get_results(
+				$wpdb->prepare(
+					"SELECT %s FROM $wpdb->posts WHERE post_parent = %d",
+					$fields,
```
Comment (trepmal) [inline revisions-cli.php]: `$fields` can't be sprintf'd into the SQL... as a string it gets quoted, as an array it throws an error.

File: revisions-cli.php [hunk 3]
```diff
@@ -76,10 +80,62 @@ public function dump( $args = array(), $assoc_args = array() ) {
 	 */
 	public function list_( $args = array(), $assoc_args = array() ) {
 
+		// Default fields to return.
+		$fields = [
```
Comment (trepmal) [inline revisions-cli.php]: possibly should make use of `\WP_CLI\Utils\get_flag_value();`

**Comment by philipjohn** [comment]
Okay, feedback addressed and tested this time ;)

**Pull request merged**